e2e/system/build_from_git_test.go

```diff
@@ -18,7 +23,66 @@ func TestBuildFromGit(t *testing.T) {
 	ctx := context.Background()
 
 	// The default image builder is kaniko here
-	kn, err := knuu.New(ctx, knuu.Options{})
+	kn, err := knuu.New(ctx, knuu.Options{TestScope: e2e.DefaultTestScope()})
+	require.NoError(t, err, "Error creating knuu")
+
+	target, err := kn.NewInstance("git-builder")
+	require.NoError(t, err, "Error creating instance")
+
+	t.Log("Building the image")
+
+	// This is a blocking call which builds the image from git repo
+	err = target.SetGitRepo(ctx, builder.GitContext{
+		Repo:     "https://github.com/celestiaorg/knuu.git",
+		Branch:   "test/build-from-git", // This branch has a Dockerfile and is protected as to not be deleted
+		Username: "",
+		Password: "",
+	})
+	require.NoError(t, err, "Error setting git repo")
+
+	t.Log("Image built")
+
+	t.Cleanup(func() {
+		if err := target.Destroy(ctx); err != nil {
+			t.Logf("Error destroying instance: %v", err)
+		}
+	})
+
+	require.NoError(t, target.Commit())
+
+	t.Logf("Starting instance")
+
+	assert.NoError(t, target.Start(ctx))
+
+	t.Logf("Instance started")
+
+	// The file is created by the dockerfile in the repo,
+	// so to make sure it is built correctly, we check the file
+	data, err := target.GetFileBytes(ctx, "/test.txt")
+	require.NoError(t, err, "Error getting file bytes")
+
+	data = []byte(strings.TrimSpace(string(data)))
+	assert.Equal(t, []byte("Hello, World!"), data, "File bytes do not match")
+}
+func TestBuildFromGitWithModifications(t *testing.T) {
```

Comment:
Why did we need to add this test to this PR?

Reply:
because we refactored how user enables Minio and Minio is required by the builder when builds on directory context and since this test modifies the built image, it does it in a tmp directory which is local.

Reply:
ah ok, thanks.